aten/src/ATen/native/cuda/Reduce.cuh

```diff
@@ -1095,7 +1095,19 @@ ReduceConfig setReduceConfig(const TensorIterator& iter){
   constexpr int min_values_per_thread = 16;
   constexpr int max_values_per_thread = 256;
 
-  if (config.values_per_thread() >= block_height * 16 || config.values_per_thread() >= max_values_per_thread) {
+  const int warp_split_threshold =
+      std::min<int>(block_height * 16, max_values_per_thread);
+  const int num_mp =
+      at::cuda::getCurrentDeviceProperties()->multiProcessorCount;
+  bool force_splitting_output = false;
+#ifdef USE_ROCM
+  force_splitting_output = iter.ndim() == 2 &&
+      reduction_on_fastest_striding_dimension &&
+      config.values_per_thread() < 1024 && num_mp < 100;
+#endif
+
+  if (!force_splitting_output &&
```

Comment:
Are you referring to this line? I see that this PR adds it and that it is always `false` in the non-ROCM case?

Reply:
Yes that's it.

Reply:
My point was that this introduced a non-`const` variable for the non-ROCM case which is practically `const`. Since there were if-defs being added anyway the condition itself could have been if-def'd to avoid adding an unused variable.

Reply:
I can clean that up in a follow up patch since this one is so close to landing.